litecommands-sponge/src/dev/rollczi/litecommands/sponge/SpongeScheduler.java

```diff
@@ -0,0 +1,59 @@
+package dev.rollczi.litecommands.sponge;
+
+import dev.rollczi.litecommands.scheduler.AbstractMainThreadBasedScheduler;
+import org.spongepowered.api.Game;
+import org.spongepowered.api.scheduler.Scheduler;
+import org.spongepowered.api.scheduler.Task;
+import org.spongepowered.plugin.PluginContainer;
+
+import java.time.Duration;
+
+public class SpongeScheduler extends AbstractMainThreadBasedScheduler {
+
+    private final PluginContainer pluginContainer;
+    private final Scheduler mainScheduler;
+    private final Scheduler asyncScheduler;
+
+    public SpongeScheduler(PluginContainer pluginContainer, Game game) {
+        this.pluginContainer = pluginContainer;
+        this.mainScheduler = findMainScheduler(game);
+        this.asyncScheduler = game.asyncScheduler();
+    }
+
+    @Override
+    public void shutdown() {
+    }
+
+    @Override
+    protected void runSynchronous(Runnable task, Duration delay) {
+        mainScheduler.submit(createTask(task, delay));
+    }
+
+    @Override
+    protected void runAsynchronous(Runnable task, Duration delay) {
+        asyncScheduler.submit(createTask(task, delay));
+    }
+
+    @Override
+    protected void log(Throwable throwable) {
+        pluginContainer.logger().error("Error completing command future:", throwable);
+    }
+
+    private Task createTask(Runnable task, Duration delay) {
+        return Task.builder()
+            .plugin(pluginContainer)
+            .execute(task)
+            .delay(delay.isPositive() ? delay : Duration.ZERO)
```

Comment:
we can put ZERO to delay method? This will work same such as without calling `delay()`?

Reply:
> we can put ZERO to delay method? This will work same such as without calling `delay()`?

yep, in fact, it is zero by default https://github.com/SpongePowered/Sponge/blob/7d3050f98615fde7c1822d9926883b5f471de127/src/main/java/org/spongepowered/common/scheduler/SpongeTask.java#L88